### PR TITLE
Feature/blog page

### PR DESCRIPTION
**Pull request by saurabh-lohiya**

### CODE REVIEW

File: index.ftd [hunk 1]
```diff
@@ -456,28 +459,46 @@ ftd.image-src src:
 optional ftd.align alignment: center
 optional ftd.resizing width:
 optional ftd.resizing height:
+optional ftd.image-src src2:
```
Comment (gsalunke) [inline index.ftd:462]:
```suggestion
optional ftd.image-src src2:
```
@saurabh-lohiya give meaningful property name, what do you mean by src2? is this the way world will use it?

File: index.ftd [hunk 2]
```diff
@@ -456,28 +459,46 @@ ftd.image-src src:
...
 width: auto
 width if { image.width != NULL }: $image.width
 max-width: fill-container
-height: auto
+height if { image.height != NULL }: $image.height
 align-self: center
 
+-- ftd.image:
+if: { image.src2 != NULL }
+src: $image.src2
```
Comment (gsalunke) [inline index.ftd:482]: @saurabh-lohiya give meaningful property name, what do you mean by src2? is this the way world will use it?

File: index.ftd [hunk 3]
```diff
@@ -1706,41 +1730,402 @@ width: fill-container
...
-    - Label 5: index.html
\ No newline at end of file
+-- component post-card-ui:
+caption title:
+string timestamp:
+ftd.image-src image: 
+string url:
+body body:
+optional common.author-meta author:
+optional string tag:
```
Comment (gsalunke) [inline index.ftd:1740]: @saurabh-lohiya is this property also present in Meenu and Priyanka's set? ask them to add it.
Also, all the new properties in other components which are not in their sets ask Meenu and Priyanka to add and you also add their sets properties for all the components.

File: index.ftd [hunk 4]
```diff
@@ -1706,41 +1730,402 @@ width: fill-container
...
+
+-- ftd.image:
+src: $post-card-ui.image
+height.fixed.px: 200
+width.fixed.px: 300
+
+-- end: ftd.row
+
+-- end: ftd.column
+
```
Comment (gsalunke) [inline index.ftd:1792]:
```suggestion
```

File: index.ftd [hunk 5]
```diff
@@ -1706,41 +1730,402 @@ width: fill-container
...
+align-content: center
+margin-top.px: 24
+
+-- ftd.text: Button
+color: $inherited.colors.cta-primary.text
+role: $inherited.types.copy-regular
+link: /
+
+-- ftd.image:
+src: $assets.files.assets.correct.svg
```
[the remark this answers is not in the document]
Comment (gsalunke) [reply index.ftd:1854]: Also rename image file name

[the remark this answers is not in the document]
Comment (gsalunke) [reply index.ftd:1854]: Also rename image file name

**Review by gsalunke — CHANGES_REQUESTED** [review]

File: index.ftd [hunk 6]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+
+-- ftd.row:
+padding-horizontal.px: 20
+padding-vertical.px: 12
+spacing.fixed.px: 8
+background.solid: $inherited.colors.cta-primary.base
+align-content: center
+margin-top.px: 24
+
+-- ftd.text: Button
```
Comment (gsalunke) [inline index.ftd:1870]:
```suggestion
-- ftd.text: $featured-post-card.cta-primary
```

File: index.ftd [hunk 7]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+
+
+
+-- component featured-post:
+common.post-meta post-data:
+optional string cta-primary: 
+optional string cta-primary-url:
+optional string cta-secondary: 
+optional string cta-secondary-url:
+optional string extra-text:
```
Comment (gsalunke) [inline index.ftd:1908]:
```suggestion
optional body body:
```

File: index.ftd [hunk 8]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+
+-- featured-post-card: $featured-post.post-data.title
+timestamp: $featured-post.post-data.published-on
+author: $featured-post.post-data.author
+body: $featured-post.post-data.body
+image: $featured-post.post-data.post-image
+extra-text: $featured-post.extra-text
```
Comment (gsalunke) [inline index.ftd:1915]:
```suggestion
```

File: index.ftd [hunk 9]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+cta-primary: $featured-post.cta-primary
+cta-primary-url: $featured-post.cta-primary-url
+cta-secondary: $featured-post.cta-secondary
+cta-secondary-url: $featured-post.cta-secondary-url
```
Comment (gsalunke) [inline index.ftd:1919]:
```suggestion
cta-secondary-url: $featured-post.cta-secondary-url

$featured-post.body
```

File: index.ftd [hunk 10]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+
+-- component newsletter:
+caption title:
+string cta-txt: Sign me up
+string cta-url: /
+string cta-placeholder: Enter your work email...
+optional body body:
+optional ftd.image-src cta-icon:
+ftd.image-src info-icon: $assets.files.static.check-circle.svg
+integer width: 1200
```
Comment (gsalunke) [inline index.ftd:2132]:
```suggestion
integer width: ????
```
@saurabh-lohiya why 1200 is needed? we have 1320 right?

File: index.ftd [hunk 11]
```diff
@@ -1706,41 +1752,604 @@ width: fill-container
...
+-- ftd.column:
+width: fill-container
+align-content: center
+
+-- ftd.row:
+width.fixed.percent: 85
+margin-vertical.px: 60
+padding-top.px: 69
+padding-bottom.px: 65
+padding-horizontal.px: 75
```
Comment (gsalunke) [inline index.ftd:2193]:
```suggestion
padding-horizontal.px: 75
```
@saurabh-lohiya padding value should not be given more than 48.

File: index.ftd (hunk 8 above)
Comment (saurabh-lohiya) [reply index.ftd:1916]: This is the property exists on priyanka's UI

File: index.ftd (hunk 7 above)
Comment (saurabh-lohiya) [reply index.ftd:1909]: body is the property which is coming from post-data

Comment (gsalunke) [reply index.ftd:1909]: @saurabh-lohiya this is not the correct format of passing `body`
```ftd
-- featured-post-card: $featured-post.post-data.title
timestamp: $featured-post.post-data.published-on
author: $featured-post.post-data.author
image: $featured-post.post-data.post-image
extra-text: $featured-post.extra-text
cta-primary: $featured-post.cta-primary
cta-primary-url: $featured-post.cta-primary-url
cta-secondary: $featured-post.cta-secondary
cta-secondary-url: $featured-post.cta-secondary-url

body: $featured-post.post-data.body
```
The correct format is:
```ftd
-- featured-post-card: $featured-post.post-data.title
timestamp: $featured-post.post-data.published-on
author: $featured-post.post-data.author
image: $featured-post.post-data.post-image
extra-text: $featured-post.extra-text
cta-primary: $featured-post.cta-primary
cta-primary-url: $featured-post.cta-primary-url
cta-secondary: $featured-post.cta-secondary
cta-secondary-url: $featured-post.cta-secondary-url

$featured-post.post-data.body
```

File: index.ftd (hunk 8 above)
Comment (gsalunke) [reply index.ftd:1915]: @saurabh-lohiya still I am not convinced why its is called as `extra-ext` can it be called as `info`?

Comment (saurabh-lohiya) [reply index.ftd:1915]: We can call that as info but priyanka will have to change that first

Comment (gsalunke) [reply index.ftd:1915]: @saurabh-lohiya change it with `info` and add comment to Priyanka's thread.

**Comment by gsalunke** [comment]
@saurabh-lohiya fix below UI review comments

1. Documentation Page:

- [ ] Remove top border of page
- [ ] Remove border of toc
- [ ] Remove border of right-column
- [ ] Remove border from toc
- [ ] Remove bottom border of page
- [ ] When borders are remove then remove the top gap also and keep moderate padding between breadcrum and page components

File: index.ftd [hunk 12]
```diff
@@ -456,30 +471,49 @@ ftd.image-src src:
...
+width if { image.width != NULL }: $image.width
+max-width: fill-container
+height if { image.height != NULL }: $image.height
+align-self: center
+
+-- end: ftd.row
+
 -- ftd.text:
 if: { image.caption != NULL }
 text: $image.caption
```
Comment (gsalunke) [inline index.ftd:505]:
```suggestion
```

File: index.ftd [hunk 13]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
-    - Label 1: index.html
-    - Label 2: index.html
-    - Label 3: index.html
-    - Label 4: index.html
-    - Label 5: index.html
+-- component post-card-ui:
+caption title:
+string timestamp:
+ftd.image-src image: 
+string url:
```
Comment (gsalunke) [inline index.ftd:1735]:
```suggestion
string link:
```

File: index.ftd [hunk 14]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+
+
+
+
+
+
+-- component featured-post-card:
+caption title:
+optional body body:
+optional string url:
```
Comment (gsalunke) [inline index.ftd:1804]:
```suggestion
optional string link:
```

File: index.ftd [hunk 15]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+
+
+
+-- component post-card:
+common.post-meta post-data:
+
+-- post-card-ui: $post-card.post-data.title
+timestamp: $post-card.post-data.published-on
+image: $post-card.post-data.post-image
+url:  $post-card.post-data.post-url
```
Comment (gsalunke) [inline index.ftd:1932]:
```suggestion
link:  $post-card.post-data.post-url
```

File: index.ftd [hunk 16]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+
+
+
+-- component post:
+common.post-meta meta:
+children atictles-container:
+optional boolean sidebar: false
+ftd.ui list right-sidebar:
+string previous-cta-txt: Back to blog
+optional string previous-cta-url:
```
Comment (gsalunke) [inline index.ftd:2030]:
```suggestion
string nav-back-text: Back to blog
optional string nav-back-link:
```

File: index.ftd [hunk 17]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+
+
+-- component newsletter-desktop:
+caption title:
+string cta-txt: 
+string cta-url: 
+string cta-placeholder: 
+optional body body:
+optional ftd.image-src info-icon: 
+ftd.color bg-color:
```
Comment (gsalunke) [inline index.ftd:2176]:
```suggestion
```
@saurabh-lohiya CSS should not be part component property

File: index.ftd [hunk 18]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+margin-vertical.px: 60
+padding-top.px: 69
+padding-bottom.px: 65
+padding-horizontal.px: 48
+background.solid: $newsletter-desktop.bg-color
+border-radius.px: 20
+align-content: center
+spacing: space-between
+
+
```
Comment (gsalunke) [inline index.ftd:2193]:
```suggestion
```

File: index.ftd [hunk 19]
```diff
@@ -1706,41 +1728,622 @@ width: fill-container
...
+-- component newsletter-mobile:
+caption title:
+string cta-txt: 
+string cta-url: 
+string cta-placeholder: 
+optional body body:
+optional ftd.image-src info-icon: 
+ftd.color bg-color:
```
Comment (gsalunke) [inline index.ftd:2266]:
```suggestion
```

File: tabs.ftd [hunk 20]
```diff
@@ -68,7 +68,7 @@ style: bold
 
 -- component tab-image:
```
Comment (gsalunke) [inline tabs.ftd:69]: @saurabh-lohiya as discussed if this component is not required - remove it.

**Review by gsalunke — CHANGES_REQUESTED** [review]

**Comment by gsalunke** [comment]
@saurabh-lohiya some UI review comments

- [ ] Light/dark mode - text colors not clearly visible - check with designer on this.
- [ ] Vertical spacing between toc-items is too much - keep it 16px

Blog Page

- [ ] Button of hero component - add 2px border and border-color should be same background-color you used there. It will help to remove flickering on mouse-over effect.

**Pull request merged**